src/init/features/apphosting/repo.ts

```diff
@@ -87,84 +91,111 @@ export async function linkGitHubRepository(
   location: string,
 ): Promise<gcb.Repository> {
   utils.logBullet(clc.bold(`${clc.yellow("===")} Set up a GitHub connection`));
-  // Create or get connection resource that contains reference to the GitHub oauth token.
-  // Oauth token associated with this connection should be used to create other connection resources.
-  let oauthConn = await getOrCreateConnection(projectId, location, APPHOSTING_OAUTH_CONN_NAME);
-  while (oauthConn.installationState.stage === "PENDING_USER_OAUTH") {
-    oauthConn = await promptConnectionAuth(oauthConn);
-  }
+  // Fetch the sentinel Oauth connection first which is needed to create further GitHub connections.
+  const oauthConn = await getOrCreateOauthConnection(projectId, location);
   const existingConns = await listAppHostingConnections(projectId);
-  if (existingConns.length < 1) {
-    const grantSuccess = await promptSecretManagerAdminGrant(projectId);
-    if (!grantSuccess) {
-      throw new FirebaseError("Insufficient IAM permissions to create a new connection to GitHub");
-    }
-    const connectionId = generateConnectionId();
-    const conn = await createConnection(projectId, location, connectionId, {
-      authorizerCredential: oauthConn.githubConfig?.authorizerCredential,
-    });
-    let refreshedConn = conn;
-    while (refreshedConn.installationState.stage !== "COMPLETE") {
-      refreshedConn = await promptAppInstall(conn);
-    }
-    existingConns.push(refreshedConn);
+
+  if (existingConns.length === 0) {
+    await ensureSecretManagerAdminGrant(projectId);
+    existingConns.push(
+      await ensureFullyInstalledConnection(projectId, location, generateConnectionId(), oauthConn),
+    );
   }
 
-  let { remoteUri, connection } = await promptRepositoryUri(projectId, existingConns);
-  while (CONFIGURE_REMOTE_URI_CHOICES.includes(remoteUri)) {
-    if (remoteUri === ADD_REPO_CHOICE) {
```

Comment:
why remove this? I like that it makes the UX simpler but would this not mean that we would be creating a new Connection whenever the user wants to make a change to the repos available to the Cloud Build Github app?

Though i'm not too sure about the consequences of having many connections ... are there any impact on billing?

Reply:
GCB `connection` resources are free so it doesn't hurt to create multiple. The idea is that once we have a Firebase GitHub app, we can dedupe `connections` on behalf of the user in our redirect handler.